Fix #11919: Some Zombified Piglins don't forgive players after death

Guilherme-Corvelo opened the pull request:
Previously, Zombified Piglins did not properly reset their aggression after a player's death, causing some to remain hostile. To fix this, an instance of the ResetUniversalAngerTargetGoal class is now an attribute of the ZombifiedPiglin class and its start() method is called upon player death(in the startPersistentAngerTimer method). This ensures that only ZombifiedPiglin entities within a range are being reset. Additionally, the expected behavior of Zombified Piglins has been updated to consistently use this reset mechanism, ensuring proper forgiveness behavior.

Video showing the problem solved:
https://www.youtube.com/watch?v=LxUmljnKg40
The video is in Portuguese, due to the fact I am still in college and doing this for a course. Sorry for the inconvenience.

Owen1212055 reviewed (changes requested):
The issue that fixes states that this is vanilla behavior and is documented on the wiki. Therefore this change will need a config option and be disabled by default.

Guilherme-Corvelo committed:
Fix #11919: Some Zombified Piglins don't forgive players after death
Previously, Zombified Piglins did not properly reset their aggression
after a player's death, causing some to remain hostile. To fix this,
an instance of the ResetUniversalAngerTargetGoal class is now an
attribute of the ZombifiedPiglin class and its start() method is
called upon player death(in the startPersistentAngerTimer method).
This ensures that only ZombifiedPiglin entities within a range are
being reset. Additionally, the expected behavior of Zombified
Piglins has been updated to consistently use this reset mechanism,
ensuring proper forgiveness behavior.

Signed-off-by: Guilherme Corvelo guilherme.corvelo@tecnico.ulisboa.pt

Guilherme-Corvelo committed:
Fix #11919: Some Zombified Piglins don't forgive players after death
Previously, Zombified Piglins did not properly reset their aggression
after a player's death, causing some to remain hostile. To fix this,
an instance of the ResetUniversalAngerTargetGoal class is now an
attribute of the ZombifiedPiglin class and its start() method is
called upon player death(in the startPersistentAngerTimer method).
This ensures that only ZombifiedPiglin entities within a range are
being reset. Additionally, the expected behavior of Zombified
Piglins has been updated to consistently use this reset mechanism,
ensuring proper forgiveness behavior.

Signed-off-by: Guilherme Corvelo guilherme.corvelo@tecnico.ulisboa.pt
